src/routers/InitializeSessionRouter.ts

```diff
@@ -15,14 +20,39 @@ class InitializeSessionRouter extends ApplicationRouter<SerializedUserSession> {
 
   async content(req: Request): Promise<SerializedUserSession> {
     const client = new OAuth2Client(process.env.GOOGLE_CLIENT_ID);
-
+    const {
+      clubs,
+      idToken,
+      graduationYear,
+      hometown,
+      interests,
+      major,
+      pronouns,
+    } = req.body;
+    const clubObjects: Club[] = await Promise.all(
+      clubs.map(async (name: string) => ClubRepo.getClubByName(name))
+    );
+    const interestObjects: Interest[] = await Promise.all(
+      interests.map(async (name: string) =>
+        InterestRepo.getInterestByName(name)
+      )
+    );
+    const majorObject = await CornellMajorRepo.getCornellMajorByName(major);
     return client
       .verifyIdToken({
-        idToken: req.body.idToken,
+        idToken,
         audience: process.env.GOOGLE_CLIENT_ID,
       })
       .then((login: LoginTicket) =>
-        UserSessionRepo.createUserAndInitializeSession(login)
+        UserSessionRepo.createUserAndInitializeSession(
+          clubObjects,
```

Comment:
Instead of passing all these elements individually, could we make an `additionalInfo` object. This way it'd be easier to add more fields in the future.

Reply:
Will be addressed in next PR.